src/main/java/com/sap/broker/budgie/helpers/AsyncOperation.java

```diff
@@ -0,0 +1,33 @@
+package com.sap.broker.budgie.helpers;
+
+import com.sap.broker.budgie.exception.InterruptedOperationException;
+
+import java.util.function.Supplier;
+
+public class AsyncOperation implements Runnable {
+
+    private int milliseconds;
+    private Supplier<AsyncOperationState> operation;
+    private AsyncOperationState state;
+
+    public AsyncOperation(Supplier<AsyncOperationState> operation, int milliseconds) {
+        this.operation = operation;
+        this.milliseconds = milliseconds;
+        new Thread(this).start();
```

Comment:
I think this is not cool.
Could you extract it outside the class and make some sort of executor for this? Or even better, introduce ExecutorService?